4.0 upgrade

wilr opened the pull request.

wilr committed:
4.0 upgrade

robbieaverill commented on composer.json:
```diff
@@ -3,21 +3,29 @@
 	"require-dev": {
-		"phpunit/PHPUnit": "~3.7@stable"
-	},
+        "phpunit/PHPUnit": "^5.7"
```
Add `squizlabs/php_codesniffer: ^3.0`

robbieaverill commented on composer.json:
```diff
@@ -3,21 +3,29 @@
+    "scripts": {
+        "lint": "phpcs src/ tests/",
+        "fix": "php-cs-fixer fix src/ && php-cs-fixer fix tests/"
+    },
```
we probably don't need to add these, or if we do they should reference a local dependency

wilr replied:
So that's just what comments has so it's consistent.

robbieaverill commented on tests/CommentNotifiableTest.php:
```diff
@@ -1,98 +1,70 @@
+        CommentNotifiableTestDataObject::class
+    ];
+
     public function setUp()
```
Let's make this protected now, tearDown too

robbieaverill commented on .travis.yml:
```diff
@@ -1,29 +1,33 @@
-      env: DB=MYSQL CORE_RELEASE=3.6
+      env: DB=MYSQL PHPUNIT_COVERAGE_TEST=1
+    - php: 7.0
+      env: DB=MYSQL
```
This won't do anything, remove

robbieaverill commented on .travis.yml:
```diff
@@ -1,29 +1,33 @@
+  - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+
+  # Bootstrap cms / mysite folder
+  - php ./cms/tests/bootstrap/mysite.php
```
I don't _think_ we need this, since it's in phpunit.xml.dist

robbieaverill reviewed (changes requested):
Nice! Minor comments. Can you also add a codecov.yml

wilr committed:
Add badges and documentation

wilr committed:
Make test functions protected.

wilr committed:
Correct dev dependancies

wilr commented on templates/SilverStripe/CommentNotifications/CommentEmail.ss:
```diff
@@ -19,7 +19,7 @@
 </ul>
 
-<blockquote>{$Comment.Comment}</blockquote>
+<blockquote><% with Comment %>$Comment<% end_with %></blockquote>
```
`$Comment.Comment` seemed to flip SSViewer out..

wilr committed:
Translate email (Fixes #2)

wilr committed:
Replace extend with invokeWithExtensions (Fixes #16)

Allows concrete commenting classes to define behaviour rather than requiring configuration to be set on an extension to the class.

wilr committed:
Remove tests not working on Travis

wilr committed:
Fixed codecov path

wilr committed:
Add coverage for onAfterPostComment

robbieaverill commented:
Replaced by #20
